crates/sats/src/algebraic_value.rs

```diff
@@ -86,11 +86,11 @@ pub enum AlgebraicValue {
     /// An [`i128`] value of type [`AlgebraicType::I128`].
     ///
     /// We box these up as they allow us to shrink `AlgebraicValue`.
-    I128(i128),
+    I128(Packed<i128>),
     /// A [`u128`] value of type [`AlgebraicType::U128`].
     ///
     /// We box these up as they allow us to shrink `AlgebraicValue`.
-    U128(u128),
+    U128(Packed<u128>),
```

Comment:
This may inhibit SIMD optimizations for i/u128. Fortunately, that doesn't matter in the slightest -- packing the types in this PR is way more important.